src/rules/preferReadonlyRule.ts

```diff
@@ -0,0 +1,193 @@
+/**
+ * @license
+ * Copyright 2017 Palantir Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as utils from "tsutils";
+import * as ts from "typescript";
+
+import * as Lint from "../index";
+import { ClassScope } from "../language/classScope";
+
+export class Rule extends Lint.Rules.TypedRule {
+    public static metadata: Lint.IRuleMetadata = {
+        description: "Requires that private variables are marked as `readonly` if they're never modified outside of the constructor.",
+        descriptionDetails: Lint.Utils.dedent`
+            If a private variable is only assigned to in the constructor, it should be declared as \`readonly\`.
+        `,
+        optionExamples: [true],
+        options: null,
+        optionsDescription: "Not configurable.",
+        rationale: Lint.Utils.dedent`
+            Marking never-modified variables as readonly helps enforce the code's intent of keeping them as never-modified.
+            It can also help prevent accidental changes of members not meant to be changed.`,
```

Comment:
In general I declare properties `readonly` if I don't want them to be modified, regardless of the access modifier.
This rule forces me to declare all private properties as `readonly` that are *currently* not modified. If later on I decide to modify such a property, I need to edit 2 locations in the file, which bloats the diff.
I can imagine my co-workers not making a change (or making a really awkward one instead), because they think the property is `readonly` on purpose and needs to stay that way. I think this may cause more confusion than good.

Reply:
> I think this may cause more confusion than good.

Funny, I've had the opposite problem: coworkers reassigning properties that were never meant to be reassigned. 

Partially to alleviate that concern, I've added an `only-inline-lambdas` option that stops checking everything except `private member = () => {}`-style arrow functions.